core/src/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -151,8 +154,8 @@ public static FileObject getFileObject( String vfsFilename, VariableSpace space,
 
       return fileObject;
     } catch ( IOException e ) {
-      throw new KettleFileException( "Unable to get VFS File object for filename '"
```

Comment:
Please move the fileObject variable in line 147 out of the try scope, then call getFriendlyURI(fileObject) instead of the string filename signature.